tests/test_dynamic_metadata.py

```diff
@@ -0,0 +1,151 @@
+import shutil
+import sys
+import zipfile
+from pathlib import Path
+from unittest import mock
+
+import git
+import pyproject_metadata
+import pytest
+
+from scikit_build_core._compat import importlib
+from scikit_build_core.build import build_wheel
+from scikit_build_core.settings.metadata import get_standard_metadata
+from scikit_build_core.settings.skbuild_read_settings import SettingsReader
+
+DIR = Path(__file__).parent.resolve()
+DYNAMIC = DIR / "packages/dynamic_metadata"
+
+
+@pytest.fixture()
+def mock_entry_points():
+    mock_version = mock.Mock()
+    mock_version.load.return_value = lambda _: "0.0.2"
+    mock_version.name = "test_version"
+    mock_version.group = "skbuild"
+    mock_version.matches.side_effect = AttributeError()
+
+    mock_readme = mock.Mock()
+    mock_readme.load.return_value = lambda _: {
+        "content-type": "text/x-rst",
+        "text": "Some text",
+    }
+    mock_readme.name = "test_readme"
+    mock_readme.group = "skbuild"
+    mock_readme.matches.side_effect = AttributeError()
+
+    mock_license = mock.Mock()
+    mock_license.load.return_value = lambda _: {"text": "MIT License"}
+    mock_license.name = "test_license"
+    mock_license.group = "skbuild"
+    mock_license.matches.side_effect = AttributeError()
+
+    def mock_ep(**_):
+        result: importlib.metadata.EntryPoints = importlib.metadata.EntryPoints(
+            (mock_version, mock_readme, mock_license)
+        )
+        return result
+
+    with mock.patch(
+        "scikit_build_core.settings.metadata.importlib.metadata.entry_points", mock_ep
+    ) as mocked:
+        yield mocked
+
+
+@pytest.mark.usefixtures("mock_entry_points")
+def test_dynamic_metadata(monkeypatch):
+    monkeypatch.chdir(DYNAMIC)
+
+    settings_reader = SettingsReader(Path("pyproject.toml"), {})
+    settings = settings_reader.settings
+
+    settings_reader.validate_may_exit()
+
+    metadata = get_standard_metadata(Path("pyproject.toml"), settings)
+
+    assert str(metadata.version) == "0.0.2"
+    assert metadata.license == pyproject_metadata.License("MIT License", None)
+    assert metadata.readme == pyproject_metadata.Readme("Some text", None, "text/x-rst")
+
+
+def test_fancy_metadata(tmp_path, monkeypatch):
+    build_dir = tmp_path / "build"
+
+    shutil.copytree(DYNAMIC, build_dir)
+    monkeypatch.chdir(build_dir)
+
+    repo = git.repo.base.Repo.init(build_dir, initial_branch="main")
```

Comment:
Needs committer info, see https://github.com/scikit-hep/cookie/blob/279a10cdb5aa1f5b71ff019ad16d9161b6bdd88e/noxfile.py#L40-L53 for example.

Reply:
Whoops, will push a fix for that.